### PR TITLE
[14.0][IMP] field vies_passed is synced from parent to children

### PR DESCRIPTION
**Pull request by EmilioPascual**
- When parent partner is updated with vies passed, children are updated.
- Process added to update partners after check vies is activated.

@moduon MT-1044 @shide @yajo plz review :)

### CODE REVIEW

File: base_vat_optional_vies/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    def execute_update_check_vies(self):
+        # Only parent partners, childs are synced from parent
```
Comment (yajo) [inline base_vat_optional_vies/models/res_config_settings.py]:
```suggestion
        # Only parent partners, children are synced from parent
```

File: base_vat_optional_vies/models/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        count_partners = self.env["res.partner"].search_count(
+            [("parent_id", "=", False)]
+        )
+        self.env["res.partner"].search([("parent_id", "=", False)]).check_vat()
+        return {
+            "effect": {
+                "fadeout": "slow",
+                "message": _("<br/>Vies passed calculated in %s partner")
```
Comment (yajo) [inline base_vat_optional_vies/models/res_config_settings.py]:
```suggestion
                "message": _("<br/>Vies passed calculated in %s partners")
```

**Review by yajo — CHANGES_REQUESTED** [review]
Some typos. No big deal.

I wanted to test on runbot, but it's failing to build:

```
Collecting vatnumber (from -r test-requirements.txt (line 11))
  Downloading vatnumber-1.2.tar.gz (19 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in vatnumber setup command: use_2to3 is invalid.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

Possibly due to the fact that the `vatnumber` dependency is unmaintained and outdated.

Could you try to replace it with https://pypi.org/project/vatnumber-xtra/ which claims to be an updated fork? Thanks!

**Review by yajo — APPROVED** [review]
Functional review good too.

Thanks for removing that dependency.

You will have to squash the commits before the merge, because the bot cannot do that by itself: https://github.com/OCA/oca-github-bot/issues/43

**Review by yajo — APPROVED** [review]

**Review by Shide — APPROVED** [review]
LGTM

**Review by yajo — APPROVED** [review]
Great! Thanks for increasing the tests coverage.

**Review by rafaelbn — APPROVED** [review]
/ocabot merge minor

**Pull request merged**